core/templates/dev/head/forms/formBuilder.js

```diff
@@ -756,6 +757,10 @@ oppia.config(['$provide', function($provide) {
         taRegisterTool(componentDefn.name, {
           display: buttonDisplay.outerHTML,
           tooltiptext: componentDefn.tooltip,
+          disabled: function() {
+            // Check canUseFs, a property on the textAngular directive scope.
+            return !this.$parent.$parent.canUseFs && componentDefn.requiresFs;
```

Comment:
Eww (to $parent.$parent). Do you think this should be a globally encapsulated in a service instead (maybe part of the explorationContextService)?